pretyping/classops.ml

```diff
@@ -298,12 +293,9 @@ let lookup_path_to_sort_from env sigma s =
   apply_on_class_of env sigma s lookup_path_to_sort_from_class
 
 let get_coercion_constructor env coe =
-  let c, _ =
-    Reductionops.whd_all_stack env Evd.empty (EConstr.of_constr coe.coe_value)
-  in
-  match EConstr.kind Evd.empty (** FIXME *) c with
-  | Construct (cstr,u) ->
-      (cstr, Inductiveops.constructor_nrealargs cstr -1)
+  match coe.coe_value with
+  | ConstructRef c ->
+      (c, Inductiveops.constructor_nrealargs c -1)
   | _ ->
```

Comment:
Here we used to reduce, so this function would still succeed on `Coercion s := S.` no? I don't know how it's used though.

Reply:
Right, this should be fixed.

Reply:
@SkySkimmer any suggestion on what is the right API to reduce in non-constructor cases?